cmd/skopeo/tags_test.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"github.com/containers/image/v5/docker/reference"
+	"testing"
+)
+
+func TestListTags(t *testing.T) {
+	// Invalid command-line arguments
+	for _, args := range [][]string{
+		{},
+		{"invalid1", "invalid2"},
+	} {
+		out, err := runSkopeo(append([]string{"list-repository-tags"}, args...)...)
+		assertTestFailed(t, out, err, "Usage:")
+	}
+
+	// invalid repository name
+	out, err := runSkopeo("list-repository-tags", "/this/doesnt/exist")
+	assertTestFailed(t, out, err, "Cannot parse repository reference")
+}
+
+func TestParsing(t *testing.T) {
+	tag := "docker.io:1000/nginx:foobar"
+	r, er2 := reference.ParseNormalizedNamed(tag)
```

Comment:
yes, the test needs a bit of work. Primarily around how to handle references that actually include a tag. In the context of a repository the tag is confusing but could just be ignored. Any opinion on if that should result in an error? Given that the command is 'list-repository-tags' if you pass a tag reference there is still a repository component that can be used, so I think silently ignoring the tag portion is viable and helps reduce the confusion of tagless docker ref implicitly being the 'latest' tag as both conditions are handled consistently.

Reply:
I’m leaning towards refusing tagged/digested input, because it indicates some probability that the user is confused about what tags/digests are, or what format the input (passed to this subcommand) is in vs. is supposed to be in. But that’s a weak opinion, I can easily see the case for accepting input that we know how to handle and possibly simplifying shell scripts (so that they don’t do something incorrect in general, like dropping anything after the last `:`)

If the goal is to test / ensure stability of parsing the input, please
- split the `repoRef` + `docker.ParseReference` code into a separate helper function
- add full input vs. output tests for _that_ function
so that if the implementation of parsing changes, the tests continue to test the updated implementation.  This test, which verifies the behavior of `reference.ParseNormalizedNamed`, which is _not called in the actual implementation at all_, does not verify / ensure stability of what the command does.

Reply:
> so that if the implementation of parsing changes, the tests continue to test the updated implementation

… and because that makes a cleanly-separated “unit” that can be cheaply unit-tested without side-effects or contacting any external registry.